src/test/blockmanager_tests.cpp

```diff
@@ -82,4 +83,122 @@ BOOST_FIXTURE_TEST_CASE(blockmanager_scan_unlink_already_pruned_files, TestChain
     BOOST_CHECK(!AutoFile(OpenBlockFile(new_pos, true)).IsNull());
 }
 
+struct FastPruneTestingSetup : public BasicTestingSetup {
+    FastPruneTestingSetup()
+        : BasicTestingSetup{CBaseChainParams::MAIN, /*extra_args=*/{"-fastprune=1"}} {}
+};
+
+BOOST_FIXTURE_TEST_CASE(blockmanager_flush_block_file, FastPruneTestingSetup)
+{
+    const auto params{CreateChainParams(ArgsManager{}, CBaseChainParams::MAIN)};
+    BlockManager blockman{{}};
+    CChain chain{};
+
+    // Ensure (by delaying execution of certain steps)
+    // that if a file is modified its timestamp will be updated.
+#ifdef WIN32
+    // FAT filesystem has write time resolution of 2 seconds
+    constexpr std::chrono::milliseconds pause(2000);
+#else
+    // EXT3 and EXT4 have nanosecond resolution
+    constexpr std::chrono::milliseconds pause(1);
+#endif
+
+    // Test blocks with no transactions, not even a coinbase
+    CBlock block1;
+    block1.nVersion = 1;
+    CBlock block2;
+    block2.nVersion = 2;
+    CBlock block3;
+    block3.nVersion = 3;
+
+    // They are 80 bytes header + 1 byte 0x00 for vtx length
+    constexpr int TEST_BLOCK_SIZE{81};
+
+    // Blockstore is empty
+    BOOST_CHECK_EQUAL(blockman.CalculateCurrentUsage(), 0);
+
+    // Write the first block; dbp=nullptr means this block doesn't already have a disk
+    // location, so allocate a free location and write it there.
+    FlatFilePos pos1{blockman.SaveBlockToDisk(block1, /*nHeight=*/1, chain, *params, /*dbp=*/nullptr)};
+
+    // Write second block
+    FlatFilePos pos2 = blockman.SaveBlockToDisk(block2, 2, chain, *params, nullptr);
```

Comment:
nit: use of assignment operator on line 126 while using initializer syntax for the same type on 123.

```suggestion
    FlatFilePos pos2{blockman.SaveBlockToDisk(block2, /*nHeight=*2, chain, *params, /*dbp=*/nullptr)};
```

Reply:
Thanks good eye. If I retouch I'll fix this.